### ISSUE TITLE
ppxml2db.py error in handling portfolioTransaction tag

### ISSUE THREAD

**Issue by Rich28** [pfalcon/ppxml2db#15]
Hi, Many thanks for developing this very useful tool.  I tried it a few months ago when it seemed to work well, but this time however  ppxml2db.py reports the 'key error' as detailed below and no data is populated into the db file. 

`/media/sf_Intel/convert to sqlite$ python3 ppxml2db.py convert_pp.xml convert_pp.db
Traceback (most recent call last):
  File "/media/sf_Intel/convert to sqlite/ppxml2db.py", line 580, in <module>
    conv.iterparse()
  File "/media/sf_Intel/convert to sqlite/ppxml2db.py", line 505, in iterparse
    assert self.uuid2ctr_map[uuid] == "portfolio"
KeyError: 'e459cc68-a149-4992-855b-1e5432befed2'`

This incidentally is with the latest versions of ppxml2db (1.0) and PP ( 0.71.2) and the XML file was saved with id attributes.  But rather than using Kommer its based on my own portfolio

**Comment by pfalcon** [comment]
Please confirm that you're using the latest git revision.

Otherwise, what I can say is that it could happen. We've seen a number of such cases over the evolution of the tool, and not all cases could have been addressed. It does work for me on files I have access to.

To resolve this issue, you would need to prepare a testcase based on your actual portfolio - first find which transactions (of all that you have) trigger that issue, then remove the rest, then obfuscate data in what remains and attach here. Finding transactions which trigger it is the hardest part, and there're no exact instructions how to do that fast. How I'd approach it is that I added debug print's around to catch which transactions trigger the error, then say remove all *after* it, which could reduce testcase size a lot. But probably still not enough. And the problem is likely not in specific transaction, but in relative order of the few of them. So, you can't just delete all *before* it.

Bottom line: if you provide a testcase, I'll try to look into it and resolve the issue. But how to provide a testcase is up to you, and you likely need to have some technical/programming experience to do that efficiently, otherwise it's just trial and error which would take a lot of time from you.

**Comment by pfalcon** [comment]
@Rich28 : I fixed a number of related issues in the latest version 1.5, can you try it? Please provide any error messages if it still occurs, but ultimate way to bust the bug would to provide an (anonymized) testcase file, like in  #16.

**Comment by pfalcon** [comment]
Assuming fixed, feel free to reopen with new info.
